sklearn/decomposition/_nmf.py

```diff
@@ -1376,3 +1544,275 @@ def inverse_transform(self, W):
         """
         check_is_fitted(self)
         return np.dot(W, self.components_)
+
+
+class MiniBatchNMF(TransformerMixin, BaseEstimator):
```

Comment:
I think MiniBachNMF could inherit from NMF which avoids re-implementing fit and tags and some of the init and maybe a bit more

Reply:
Ok, this will need some more work... I'll keep you posted... :)

Reply:
@jeremiedbb , I've tried to inherit `MiniBatchNMF` from `NMF`: indeed this allows to avoid re-implementing `fit` and `tags`.
`partial_fit` should be defined anyway. `fit_transform` needs to be redefined. Still, inheriting from NMF forces me to redefine `transform` which I don't think is needed in the online version of the algorithm. Am I missing something? Thanks for any insight.